src/apps/relay/dbdrivers/dbd_redis.c

```diff
@@ -163,9 +167,6 @@ static Ryconninfo *RyconninfoParse(const char *userdb, char **errmsg) {
     if (!(co->host)) {
       co->host = strdup("127.0.0.1");
     }
-    if (!(co->password)) {
```

Comment:
Why this change?

Reply:
Originally during testing I created an extra branch `if (!(co->user)) { co->user = strdup(""); }`.

However, I found out that it made coturn always authenticate with a username and password, which would break for redis servers that don't have the ACL feature.

That's because the branches using password checked if the pointer to password was set, which it always was with the statement I just removed, rather than if password contained a useful value. So the password conditionals haven't been working for a while, coturn just discarded the error that redis threw out and continued on (which is fine, but not ideal).

Hence also changes to make those branches `co->password && strlen(co->password)` rather than just `co->password`, which I could effectively change to `co->password && co->password[0]` but strlen felt nicer.

Error handling of missing auth doesn't seem to be handled fully correctly, but happens moreso as a side effect, the side effect of using the side effect is it never warns about if the password is needed which is why there are no logs complaining about use of `AUTH` when wasn't needed.